Objects/longobject.c

```diff
@@ -169,20 +169,17 @@ _PyLong_New(Py_ssize_t size)
 }
 
 PyLongObject *
-_PyLong_FromDigits(int negative, Py_ssize_t digit_count, digit *digits)
+_PyLong_FromDigits(int negative, Py_ssize_t digit_count, Py_digit *digits)
```

Comment:
Now this helper is unused in pr, so I suggest to revert that.  Added code just runs some useless checks, etc.

On another hand, new writer API could be used in dec_as_long().  This is an opportunity to hide this helper, even inline one in _PyLong_Copy().

Reply:
I restored the original _PyLong_FromDigits() to leave it unchanged.